File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{
    public static double SuccessRate(int speed) => speed switch
    {
        < 1 => 0.0,
        < 5 => 1.0,
        < 9 => 0.9,
        < 10 => 0.8,
        _ => 0.77
    };

    private const double CarsProducedPerHour = 221.0;

    public static double ProductionRatePerHour(int speed) =>
        speed * CarsProducedPerHour * SuccessRate(speed);

    public static int WorkingItemsPerMinute(int speed) => (int)(ProductionRatePerHour(speed) / 60);
}

File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {
    private static final double BASE_RATE = 221.0;

    public double productionRatePerHour(int speed) {
        return speed * BASE_RATE * successRate(speed);
    }

    public int workingItemsPerMinute(int speed) {
        return (int) (productionRatePerHour(speed) / 60);
    }

    private static double successRate(int speed) {
        return switch (speed) {
            case 10 -> 0.77;
            case 9 -> 0.8;
            case 5, 6, 7, 8 -> 0.9;
            default -> 1.0;
        };
    }
}
